consensus/bor/contract/client.go

```
@@ -1,66 +1,51 @@
-	stateReceiverABI = `[{"constant":true,"inputs":[],"name":"SYSTEM_ADDRESS","outputs":[{"internalType":"address","name":"","type":"address"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"lastStateId","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":false,"inputs":[{"internalType":"uint256","name":"syncTime","type":"uint256"},{"internalType":"bytes","name":"recordBytes","type":"bytes"}],"name":"commitState","outputs":[{"internalType":"bool","name":"success","type":"bool"}],"payable":false,"stateMutability":"nonpayable","type":"function"}]`
-)
+func NewGenesisContractsClient(chainConfig *params.ChainConfig, validatorSetAddr, stateReceiverAddr common.Address) *GenesisContractsClient {
+	backend, err := ethclient.Dial("127.0.0.1")
+	if err != nil {
+		log.Error("Failed to initialize ethereum client", "error", err)
```

could we return this errors and check them then?